src/prefect/environments/storage/s3.py

```diff
@@ -78,9 +79,15 @@ def get_flow(self, flow_location: str) -> "Flow":
         self.logger.info("Downloading {} from {}".format(flow_location, self.bucket))
 
         # Download stream from S3
-        self._boto3_client.download_fileobj(
-            Bucket=self.bucket, Key=flow_location, Fileobj=stream
-        )
+        from botocore.exceptions import ClientError
+
+        try:
+            self._boto3_client.download_fileobj(
+                Bucket=self.bucket, Key=flow_location, Fileobj=stream
+            )
+        except ClientError as err:
+            self.logger.info("Error downloading Flow from S3: {}".format(err))
```

Comment:
two questions:
- if we log this it should be `error`
- I'm worried that returning `None` just causes a more cryptic error downstream, especially when flows are run in Cloud; wouldn't it be more transparent to let the ClientError surface so that the flow run is marked as failed?

Reply:
Oh yeah that should be `error`. And okay I'm fine with surfacing the client error 👍